### PR TITLE
stuff

### LINKED ISSUE #92: keybinds
maybe for:
* jumping to search box
* reloading
* account tab
* settings page
* home (from acc/settings page)
* share

also make `ctrl+/` the keybind to list

### LINKED ISSUE #193: sorting options for comment feeds
sort by newest, oldest, most likes, and most quotes

should be configurable on settings page and on post pages, and saved in localstorage

### LINKED ISSUE #228: require password to set/change email
could be on the "email verified" page saying something like "enter your current password to finish email verification"

### LINKED ISSUE #235: make showlog not just be a div
could be a bubble that comes in from the top that's fixed position or something like that

### LINKED ISSUE #246: "recent" option on hashtags tl

### LINKED ISSUE #260: dynamic favicons are broken with auto theme

### LINKED ISSUE #264: add a way to get IPs using a reverse proxy
add a reverse proxy option and any time you need an ip itll check for a header if it's enabled

### LINKED ISSUE #265: better docs
add docs for
- createsuperuser
- upgrading instance w scripts
- etc

### LINKED ISSUE #266: make usernames case insensitive
across most of the site they are, however there are a few places where they aren't (ex. badge add/remove), so that needs to be fixed

### LINKED ISSUE #270: sitemaps for hashtag pages

### LINKED ISSUE #272: auto database backups
there should be config for the path/filename, frequency (days per backup), and how many backups to keep

### LINKED ISSUE #274: Followers feed is broken for my account
Reproduction steps: Log into my account on smiggins, switch to followers feed, and see that you get 500 Errors. I think that this might be related to missing posts or accounts that my followers list in particular would try to grab, and error upon finding them missing.

### LINKED ISSUE #275: overhaul pronoun system
it should be per-language due to differences in how languages work. possible format:
```jsonc
{
  "pronouns": {
    "primary": {
      "f": "she/her",
      "x": {
        "name": "Unset",
        "special": "no-secondary"
      },
      // etc
    },
    "secondary": {
      "f": "she/her",
      "i": {
        "name": "Inherit",
        "special": "inherit"
      },
      // etc
    },
    "visible": {
      "x": "Unset",
      "ff": "she/her",
      // etc
    }
  }
}
```

### LINKED ISSUE #276: make `re: `s in cws not compound

### LINKED ISSUE #277: add an option to always expand content warnings

### PR DESCRIPTION
**Pull request by trinkey**
## What does this do?
small changes

## Issues completed:
- fixes #266
- fixes #264
- fixes #228
- fixes #270 
- fixes #274 
- fixes #265 
- fixes #272 
- fixes #275 
- fixes #260 
- fixes #276 
- fixes #92 
- fixes #193 
- fixes #246 
- fixes #235 
- fixes #277 

---
### Checklist for Reviewers
- [x] Version has been incremented (if appropriate) on the `REAL_VERSION` variable in `backend/variables.py`
- [ ] Any new changes have been tested thoroughly (including api calls that wouldn't normally be possible using inspect element or the /api/docs page)
- [x] Any changes have been documented in `CHANGELOG.md`
- [ ] Features that break easily have been tested (mostly viewing pages when not logged in)

### CODE REVIEW

**Review by Subroutine7901 — APPROVED** [review]
All of the changes look good, I haven't done too much testing but I manually reviewed all of the code.

**Comment by Subroutine7901** [comment]
Wait! Never mind I forgot about the things I commented on in the review, act like I requested changes, there's a few typos...

File: docs/contributing.md
[the remark this answers is not in the document]
Comment (trinkey) [reply docs/contributing.md:1]: false it's just like that

**Review by Subroutine7901 — APPROVED** [review]
*Now* everything looks good. @Sodiboo did think that the list of lang codes in the translation docs should be replaced with a more accurate resource, but that can wait I guess.

**Comment by Subroutine7901** [comment]
Also can I press the button..?

**Comment by Subroutine7901** [comment]
image

**Comment by trinkey** [comment]
> Also can I press the button..?

fiiinne

**Comment by trinkey** [comment]
> @sodiboo did think that the list of lang codes in the translation docs should be replaced with a more accurate resource, but that can wait I guess.

find one and ill put it

**Comment by Subroutine7901** [comment]
> > @sodiboo did think that the list of lang codes in the translation docs should be replaced with a more accurate resource, but that can wait I guess.
> 
> find one and ill put it

:3 (also I'll tell you if I find one, and will probably just commit it myself...)

**Comment by sodiboo** [comment]
ISO 639-3, i guess.

https://en.m.wikipedia.org/wiki/ISO_639-3

e.g.
https://en.m.wikipedia.org/wiki/ISO_639:e

**Comment by Subroutine7901** [comment]
Yea like https://en.m.wikipedia.org/wiki/List_of_ISO_639-3_codes

**Comment by trinkey** [comment]
i like -1 codes better but idk

https://en.m.wikipedia.org/wiki/List_of_ISO_639_language_codes

**Comment by sodiboo** [comment]
> i like -1 codes better but idk
> 
> https://en.m.wikipedia.org/wiki/List_of_ISO_639_language_codes

yeah but they're less extensive. they literally cover less languages. notably, missing toki pona, which is the translation I started work on. ISO 639-3 includes toki pona.

if you prefer 639-1 codes, then you can say that your languages should be named after 639-1 if one exists, otherwise 639-3. the filenames are just arbitrary anyway. it doesn't actually matter which system you use in the end, as long as all relevant languages have a unique filename.

**Comment by trinkey** [comment]
> if you prefer 639-1 codes, then you can say that your languages should be named after 639-1 if one exists, otherwise 639-3. the filenames are just arbitrary anyway. it doesn't actually matter which system you use in the end, as long as all relevant languages have a unique filename.

i think this is good i can update docs eventually

**Pull request merged**